sdk/core/Azure.Core/tests/ModelSerializationTests/Animal.cs

```diff
@@ -0,0 +1,164 @@
+
+        internal Animal(string name)
+        {
+            Name = name;
+        }
+
+        #region Serialization
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer, SerializableOptions options)
+        {
+            writer.WriteStartObject();
+            if (options.IgnoreReadOnlyProperties)
+            {
+                writer.WritePropertyName("latinName"u8);
+                writer.WriteStringValue(LatinName);
+            }
+            writer.WritePropertyName("name"u8);
+            writer.WriteStringValue(Name);
+            writer.WritePropertyName("isHungry"u8);
+            writer.WriteBooleanValue(IsHungry);
+            writer.WritePropertyName("weight"u8);
+            writer.WriteNumberValue(Weight);
+
+            if (options.IgnoreAdditionalProperties)
+            {
+                //write out the raw data
+                foreach (var property in RawData)
+                {
+                    writer.WritePropertyName(property.Key);
+#if NET6_0_OR_GREATER
+                    writer.WriteRawValue(property.Value);
+#else
+                    JsonSerializer.Serialize(writer, JsonDocument.Parse(property.Value.ToString()).RootElement);
+#endif
+                }
+            }
+            writer.WriteEndObject();
+        }
+
+        internal static Animal DeserializeAnimal(JsonElement element, SerializableOptions options)
+        {
+            double weight = default;
+            string name = "";
+            string latinName = "";
+            bool isHungry = default;
+
+            Dictionary<string, BinaryData> rawData = new Dictionary<string, BinaryData>();
+            foreach (var property in element.EnumerateObject())
+            {
+                if (property.NameEquals("weight"u8))
+                {
+                    weight = property.Value.GetDouble();
+                    continue;
+                }
+                if (property.NameEquals("name"u8))
+                {
+                    name = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("latinName"u8))
+                {
+                    latinName = property.Value.GetString();
+                    continue;
+                }
+                if (property.NameEquals("isHungry"u8))
+                {
+                    isHungry = property.Value.GetBoolean();
+                    continue;
+                }
+
+                if (options.IgnoreAdditionalProperties)
+                {
+                    //this means it's an unknown property we got
+                    rawData.Add(property.Name, BinaryData.FromString(property.Value.GetRawText()));
+                }
+            }
+            return new Animal(weight, latinName, name, isHungry, rawData);
+        }
+        #endregion
+
+        #region InterfaceImplementation
+        public bool TryDeserialize(Stream stream, out long bytesConsumed, SerializableOptions options = default)
```

Comment:
Once the design is settled, will there be follow-up PR for `autorest.csharp` to generate the implementation for all ResourceData which implements `IJsonSerializable`?